libclc/clc/include/clc/clcmacro.h

```diff
@@ -207,10 +207,6 @@
     return BUILTIN(x);                                                         \
   }
 
-#ifdef cl_khr_fp16
-
-#pragma OPENCL EXTENSION cl_khr_fp16 : enable
-
 #define _CLC_DEFINE_UNARY_BUILTIN_FP16(FUNCTION)                               \
   _CLC_DEF _CLC_OVERLOAD half FUNCTION(half x) {                               \
     return (half)FUNCTION((float)x);                                           \
```

Comment:
This is weird because we still have the reference to half in the header. Should the header just : disable it at the end?

Reply:
True. It wouldn't solve the problem of other headers potentially disabling the pragmas, so we'd still need (or prefer) users to enable pragmas at the "call sites". But as a matter of principle I'm happy to do as you suggest.

Reply:
Another thought would be that we make it generic as a macro that calls a unary/binary builtin casting type `T` to type `U` and require users to pass `half` and `float`.